### ISSUE TITLE
Puppet-lint does not report hash alignment

### ISSUE THREAD

**Issue by shawark** [rodjek/puppet-lint#414]
As title described, `puppet-lint` does not report alignment error, for example.

```
$myhash = { key                => "some value",
                other_key => "some other value" }
```

According to https://docs.puppetlabs.com/guides/style_guide.html#spacing-indentation-and-whitespace, `=>` should be aligned.

**Comment by rothsa** [comment]
I noticed you used the example here for hashes: https://docs.puppetlabs.com/puppet/latest/reference/lang_datatypes.html. (Getting their particular example to pass puppet-lint, when it should not meant changing the double quotes to single, I could check with someone at puppet about fixing that...)

The notes there say `Should align hash rockets (=>) within blocks of attributes, remembering to arrange hashes for maximum readability first.`...so I guess one would need to figure out precisely when a hash passes outside of "maximum readability" in order to have puppet-lint check for that. 

Since they don't define it, would you agree that a loose warning when neither the key, nor the arrow alignment is adequate to give coverage to this case? I've started a fork towards that idea.

**Comment by shawark** [comment]
I am personally happy with a loose warning.
However why not reuse the codes of checking resource hash alignment?

**Comment by ferventcoder** [comment]
I am a fan of leaving this as is, mostly because of the [ACL](https://github.com/puppetlabs/puppetlabs-acl) type.

```puppet
permissions => [
     { identity => 'Administrators', rights => ['full'] },
     { identity => 'Administrator', rights   => ['full'], perm_type => 'allow', child_types => 'all', affects => 'all' }
   ],
```

You can see how this could get very chatty if you had a few more of these specified. To be clear, it's the more than one hash per line.

**Comment by shawark** [comment]
I guess you already a lot of "over 80 characters" warnings which you choose to ignore, so you can do the same to hash alignment warnings :smile:

**Comment by rothsa** [comment]
@shawark sure, I'll look into that. :+1: 

This could easily get out of hand in complexity, but a roll-up of each type of warning makes sense to me. 

It's nice to know when there's a number of warnings, but be able to scan for the more important stuff without without scrolling or searching for them.

**Comment by ferventcoder** [comment]
@shawark fair enough ;)

**Comment by rnelson0** [comment]
There is apparently no checking of alignment, warning or error, inside a hash definition. This needs investigation.

**Comment by rnelson0** [comment]
Having done some belated research on this, this is by design. If someone is willing to create a new check that is disabled by default to catch this, it would probably be accepted. Thanks!
